# HG changeset patch
# User Chaiwat Suttipongsakul <cwt@bashell.com>
# Date 1533402042 -25200
# Node ID 509903f8adb1ba70c6f72d411af5d48d946d215b
# Parent  bbad9283f80f733f39025cab93c6ec7bfa12434f
support for IGNORES as a set in hg 4.7

diff --git a/hggit/__init__.py b/hggit/__init__.py
--- a/hggit/__init__.py
+++ b/hggit/__init__.py
@@ -73,9 +73,14 @@
     # baseset was added in hg 3.0
     pass
 
-demandimport.ignore.extend([
-    'collections',
-])
+if hasattr(demandimport, 'ignore') and isinstance(demandimport.ignore, list):
+    demandimport.ignore.extend([
+        'collections',
+    ])
+elif hasattr(demandimport, 'IGNORES') and isinstance(demandimport.IGNORES, set):
+    demandimport.IGNORES.update({
+        'collections',
+    })
 
 __version__ = '0.8.11'
 
# HG changeset patch
# User Chaiwat Suttipongsakul <cwt@bashell.com>
# Date 1533469473 -25200
# Node ID c49e61d147523a45c51166ab8e429e3e8ea364b4
# Parent  509903f8adb1ba70c6f72d411af5d48d946d215b
use hgutil.safehasattr instead of hasattr

diff --git a/hggit/__init__.py b/hggit/__init__.py
--- a/hggit/__init__.py
+++ b/hggit/__init__.py
@@ -73,11 +73,11 @@
     # baseset was added in hg 3.0
     pass
 
-if hasattr(demandimport, 'ignore') and isinstance(demandimport.ignore, list):
+if hgutil.safehasattr(demandimport, 'ignore'):
     demandimport.ignore.extend([
         'collections',
     ])
-elif hasattr(demandimport, 'IGNORES') and isinstance(demandimport.IGNORES, set):
+elif hgutil.safehasattr(demandimport, 'IGNORES'):
     demandimport.IGNORES.update({
         'collections',
     })
